langwatch/src/pages/settings/members/[userId].tsx

```diff
@@ -0,0 +1,262 @@
+          </Link>
+        </VStack>
+      </SettingsLayout>
+    );
+  }
+
+  return (
+    <SettingsLayout>
+      <VStack
+        paddingX={4}
+        paddingY={6}
+        gap={6}
+        width="full"
+        maxWidth="980px"
+        align="start"
+      >
+        <HStack gap="8px">
+          <Link href="/settings/members">Members</Link>
+          <Icon>
+            <ChevronRight width={12} />
+          </Icon>
+          <Text>{member.user.name}</Text>
+        </HStack>
+
+        <HStack width="full" justify="space-between" align="center">
+          <Heading size="lg" as="h1">
+            Member
+          </Heading>
+        </HStack>
+
+        <Card.Root width="full">
+          <Card.Body paddingY={2}>
+            <HorizontalFormControl label="Email">
+              <Text>{member.user.email}</Text>
+            </HorizontalFormControl>
+            <HorizontalFormControl
+              label="Organization Role"
+              helper="Team-specific roles are set per team below."
+            >
+              <Field.Root>
+                <OrganizationUserRoleField
+                  organizationId={organization.id}
+                  userId={member.userId}
+                  defaultRole={member.role}
+                />
+              </Field.Root>
+            </HorizontalFormControl>
+          </Card.Body>
+        </Card.Root>
+
+        <Heading size="md">Teams</Heading>
+        <Card.Root width="full">
+          <Card.Body paddingY={0} paddingX={0}>
+            <Table.Root variant="line" width="full">
+              <Table.Header>
+                <Table.Row>
+                  <Table.ColumnHeader>Team</Table.ColumnHeader>
+                  <Table.ColumnHeader w={"35%"}>Role</Table.ColumnHeader>
+                  <Table.ColumnHeader w={"10%"}>Actions</Table.ColumnHeader>
+                </Table.Row>
+              </Table.Header>
+              <Table.Body>
+                {member.user.teamMemberships
+                  .filter((tm) => tm.team.organizationId === organization.id)
+                  .map((tm) => (
+                    <Table.Row key={tm.team.id}>
+                      <Table.Cell>
+                        <Link href={`/settings/teams/${tm.team.slug}`}>
+                          {tm.team.name}
+                        </Link>
+                      </Table.Cell>
+                      <Table.Cell>
+                        <Field.Root>
+                          <TeamUserRoleField
+                            member={{
+                              ...tm,
+                              user: member.user,
+                            }}
+                            organizationId={organization.id}
+                            customRole={(() => {
+                              const customRoleAssignment =
+                                member.user.customRoleAssignments.find(
+                                  (cra) => cra.teamId === tm.teamId,
+                                );
```

Comment:
This runs on every render and creats a new object, forcing a rerender for all children too (I think?)